js_modules/dagit/packages/core/src/partitions/usePartitionStepQuery.tsx

```diff
@@ -0,0 +1,238 @@
+import {gql, useApolloClient, ApolloClient} from '@apollo/client';
+import * as React from 'react';
+
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorInfo';
+import {PythonErrorFragment} from '../app/types/PythonErrorFragment';
+import {DagsterTag} from '../runs/RunTag';
+import {RunFilterToken} from '../runs/RunsFilterInput';
+import {RunStatus} from '../types/globalTypes';
+
+import {PartitionMatrixStepRunFragment} from './types/PartitionMatrixStepRunFragment';
+import {
+  PartitionStepLoaderQuery,
+  PartitionStepLoaderQueryVariables,
+} from './types/PartitionStepLoaderQuery';
+import {PartitionRuns, PARTITION_MATRIX_STEP_RUN_FRAGMENT} from './useMatrixData';
+
+interface DataState {
+  runs: PartitionMatrixStepRunFragment[];
+  partitionNames: string[];
+  loading: boolean;
+  loadingCursorIdx: number;
+  error?: PythonErrorFragment;
+}
+
+const InitialDataState: DataState = {
+  runs: [],
+  partitionNames: [],
+  loading: false,
+  loadingCursorIdx: 0,
+};
+
+/**
+ * This React hook mirrors `useCursorPaginatedQuery` but collects each page of partitions
+ * in slices that are smaller than pageSize and cause the results to load incrementally.
+ */
+export function usePartitionStepQuery(
```

Comment:
Could this replace `useChunkedPartitionsQuery`? It looks like it does some of the same stuff, but with new options?

Reply:
yeah, the aim is to replace `useChunkedPartitionsQuery`.  It should do the same stuff, but with different options and a lighter payload of data fetching.